modules/ingester/flush.go

```diff
@@ -124,16 +165,38 @@ func (i *Ingester) flushLoop(j int) {
 		}
 		op := o.(*flushOp)
 
-		level.Debug(log.Logger).Log("msg", "flushing block", "userid", op.userID, "fp")
-
-		err := i.flushUserTraces(op.userID)
-		if err != nil {
-			level.Error(log.WithUserID(op.userID, log.Logger)).Log("msg", "failed to flush user", "err", err)
-
-			// re-queue failed flush
-			op.from += int64(flushBackoff)
-			i.flushQueues.Requeue(op)
-			continue
+		if op.kind == opKindComplete {
+			level.Debug(log.Logger).Log("msg", "completing block", "userid", op.userID)
+			instance, exists := i.getInstanceByID(op.userID)
+			if !exists {
+				// instance no longer exists? that's bad, log and continue
+				level.Error(log.Logger).Log("msg", "instance not found", "tenantID", op.userID)
+				continue
+			}
+
+			isComplete := instance.CompleteBlock()
```

Comment:
after dequeue is called the caller must either call `Requeue()` or `Clear()` to signal success or failure.

we need to return an error from `instance.CompleteBlock()` so we know which one to do.

Reply:
So about this, I was wondering if we should retry this at all. It seems like it could only do more harm... like use up all the space on disk. wdyt?

Reply:
yes, we should retry.  we just need to make sure it cleans up after itself correctly on failure.  for now at least this is going to be a very manual process that we watch after calling `/shutdown`.  so if there are errors we will be there to intervene.